.gitignore

```diff
@@ -39,6 +39,7 @@
 /dev/integration_tests/**/Pods
 /packages/flutter/coverage/
 version
+.version.json
```

Comment:
@zanderso what do you think? Having a dirty checkout of the framework would cause `flutter upgrade` to fail, so I think it would make sense to cherrypick this. If not, then we should probably land the gitignore change first, and then only land the rest of this once that's reached stable.

Reply:
Can the `.version.json` live in a directory that's already `.gitignore`'d? Maybe `bin/cache`?

Reply:
that's a good idea!

Reply:
Seems like moving into the cache directory is causing errors when trying to find the version file.

We may need to first create the version file recursively before we try to write to it now that it is in a nested directory?